src/tools/schema2base.ts

```diff
@@ -16,8 +16,7 @@ import {SchemaGraph, SchemaNode} from './schema2graph.js';
 import {ParticleSpec} from '../runtime/particle-spec.js';
 
 export interface ClassGenerator {
-  addField(field: string, typeChar: string): void;
-  addReference(field: string, refName: string): void;
+  addField(field: string, typeChar: string, isOptional: boolean, refClassName: string|null): void;
```

Comment:
Optional: there's enough args here now that this might be nicer with an options argument instead.

Reply:
Possibly, don't think I'll bother at this point.